src/rgw/rgw_op.cc

```diff
@@ -4843,6 +4843,16 @@ void RGWPutLC::execute()
   RGWLCXMLParser_S3 parser(s->cct);
   RGWLifecycleConfiguration_S3 new_config(s->cct);
 
+  content_md5 = s->info.env->get("HTTP_CONTENT_MD5");
+  if (content_md5 == nullptr) {
```

Comment:
is AWS s3 behaviour similar, do they reject requests without a HTTP_CONTENT_MD5 for lifecycle...  not sure whether we should error out if no md5 is supplied

Reply:
@theanalyst Yeah, if not supplying the `Content-MD5` in request headers, the response message of AWS S3 is
`<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>InvalidRequest</Code><Message>Missing required header for this request: Content-MD5</Message>
...
</Error>`